### DISCUSSION TITLE
Is it possible to expose per-contact frictional forces in PhysicsDirectBodyState3D?

### DISCUSSION THREAD

**Discussion by Not-Parker** [godot-jolt/godot-jolt#790] [Questions] [answered]
This is more of a question rather than a feature request because I'm not sure if it's possible due to Godot's current api.

I need to be able to have more control over when friction is applied to a rigid body per contact, and the easiest way I could think of was to expose the force applied / change in velocity due to friction, similar to PhysicsDirectBodyState3D.GetContactImpulse(). With this I could just subtract this velocity from the linear velocity (in my case rotation is disabled which makes this easier). Is this something that would be possible at some point? Are there any other ways to achieve this with the current api?

**Comment by mihe** [comment, marked as answer]
At this point, with the way the things are looking in terms of integrating this extension as an official engine module at some point, I'm strictly adhering to the physics server interface as exposed by Godot, which does not provide anything like this.

Jolt does however provide this data as part of the method I'm relying on for the contact impulse (see [here](https://github.com/jrouwe/JoltPhysics/blob/c55ce1c8d5b4ade2d8e69d3c1fbeec71383db798/Jolt/Physics/Collision/EstimateCollisionResponse.h#L11-L32)), but your best bet is probably to make a Godot proposal for exposing this kind of stuff.
